src/Symfony/Bundle/FrameworkBundle/Command/SecretsAddCommand.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Exception\EncryptionKeyNotFoundException;
+use Symfony\Bundle\FrameworkBundle\Secret\MutableSecretStorageInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class SecretsAddCommand extends Command
+{
+    protected static $defaultName = 'secrets:add';
+
+    private $secretsStorage;
+
+    public function __construct(MutableSecretStorageInterface $secretsStorage)
+    {
+        $this->secretsStorage = $secretsStorage;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Adds a secret in the local storage.')
+            ->addArgument('name', InputArgument::REQUIRED)
+            ->addArgument('secret', InputArgument::REQUIRED)
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $name = $input->getArgument('name');
+        $secret = $input->getArgument('secret');
+
+        try {
+            $this->secretsStorage->putSecret($name, $secret);
+        } catch (EncryptionKeyNotFoundException $e) {
+            throw new \LogicException(sprintf('No encryption keys found. You should call the "%s" command.', SecretsGenerateKeyCommand::getDefaultName()));
+        }
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $name = $input->getArgument('name');
+        while (empty($name)) {
+            $name = $io->askQuestion(new Question('Name of the secret'));
+            if (empty($name)) {
+                $io->warning('The "name" cannot be empty.');
+            }
+        }
+        $input->setArgument('name', $name);
+
+        $secret = $input->getArgument('secret');
+        if (empty($secret)) {
+            $question = new Question('Plain text secret value', $input->getArgument('secret'));
```

Comment:
Maybe you can simplify this code using `$io->askHidden()`

Reply:
indeed